frontend/linux/rpm/distro/dnf_install.go

```diff
@@ -0,0 +1,263 @@
+package distro
+
+import (
+	"context"
+	"fmt"
+	"path/filepath"
+	"strings"
+
+	"github.com/Azure/dalec"
+	"github.com/Azure/dalec/frontend"
+	"github.com/moby/buildkit/client/llb"
+	gwclient "github.com/moby/buildkit/frontend/gateway/client"
+)
+
+var dnfRepoPlatform = dalec.RepoPlatformConfig{
+	ConfigRoot: "/etc/yum.repos.d",
+	GPGKeyRoot: "/etc/pki/rpm-gpg",
+	ConfigExt:  ".repo",
+}
+
+type PackageInstaller func(*dnfInstallConfig, string, []string) llb.RunOption
+
+type dnfInstallConfig struct {
```

Comment:
We only have a dnf install config and not a tdnfInstallConfig, since in Dalec thus far every option we use is both dnf and tdnf compatible. In the future as we add more distros, we may need to rethink this portion, but if we stick to only using the options that are the same between the two, then it makes our life easier!